### PR TITLE
Add "JavaScript Start-up Optimization" (Fixes #4924)

### LINKED ISSUE #4924: Primer on efficiently loading & bundling JavaScript
We don't currently have any guidance on techniques like smarter bundling / code-splitting / only shipping the minimal JS you need and deferring the rest till later. Aspects of this are touched on in the PRPL pattern, but we don't really do a deep dive on this topic. 

We might also want to touch on patterns here like [route-based chunking](https://gist.github.com/addyosmani/44678d476b8843fd981ff8011d389724) that similarly don't have any content on /web but are quite related.

### PR DESCRIPTION
**Pull request by addyosmani**
Reviewers: @PaulKinlan @petele

### CODE REVIEW

File: src/content/en/fundamentals/performance/optimizing-content-efficiency/javascript-startup-optimization/index.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
+project_path: /web/fundamentals/_project.yaml
+book_path: /web/fundamentals/_book.yaml
+description: You should keep your network transmission and parse/compile cost
```
Comment (petele) [inline src/content/en/fundamentals/performance/optimizing-content-efficiency/javascript-startup-optimization/index.md]: `description` fields need to be on a single line

Comment (addyosmani) [reply src/content/en/fundamentals/performance/optimizing-content-efficiency/javascript-startup-optimization/index.md]: Thanks for the confirm. I'll get that updated.

On your other comment about review from a TW, it's been reviewer by 3 technical reviewers but not a writer just yet. Perhaps @kaycebasques might have time for it.

**Comment by kaycebasques** [comment]
I'm reviewing now

**Comment by petele** [comment]
@kaycebasques - once you're happy, please go ahead and merge!

File: src/content/en/fundamentals/performance/optimizing-content-efficiency/javascript-startup-optimization/index.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+engine to **parse/compile** this code. In [Chrome
+DevTools](web/tools/chrome-devtools/), parse and compile are part of the yellow
+"Scripting" time in the Performance panel.
+
+<img src="images/1__4gNDmBlXxOF2-KmsOrKkw.png"/>
+
+The Bottom-Up/Call Tree allow viewing exact Parse/compile timings:
+
+<figure> <img src="images/1_GdrVt_BTTzzBOIoyZZsQZQ.png"/> <figcaption> Chrome
+DevTools Performance panel > Bottom-Up. With V8’s Runtime Call Stats enabled, we
```
[the remark this answers is not in the document]
Comment (addyosmani) [reply src/content/en/fundamentals/performance/optimizing-content-efficiency/javascript-startup-optimization/index.md]: Good call. I'll add in that note. Locally addressed the rest of your feedback which should be in the next set of commits soon.

**Comment by addyosmani** [comment]
@kaycebasques Over to you for feedback review.

**Comment by kaycebasques** [comment]
Sorry, didn't realize you were blocked on me!

**Comment by kaycebasques** [comment]
@petele I can't merge...

screen shot 2017-12-13 at 8 24 19 am

**Comment by addyosmani** [comment]
@petele @PaulKinlan Looks like I can't merge this either :) In the new year, would supes appreciate if someone with access could merge this.

**Comment by kaycebasques** [comment]
@petele the CLA bot is preventing us from merging this

**Comment by petele** [comment]
Meep - sorry about the CLA bot, merging now!

**Pull request merged**